File: backend/GeoQuiz_backend/GeoQuiz_backend/Application/DTOs/KingOfTheHill/RoundStartedData.cs
using GeoQuiz_backend.Application.DTOs.PvP;
using GeoQuiz_backend.Application.Payloads.Questions;
using GeoQuiz_backend.Domain.Entities;
using GeoQuiz_backend.Domain.Mongo;

namespace GeoQuiz_backend.Application.DTOs.KingOfTheHill
{
    public class RoundStartedData
    {
        public int RoundNumber { get; set; }
        public int RoundType { get; set; } 
        public QuestionData Question { get; set; } = null!;
        public long ServerTime { get; set; }
        public long RoundEndAt { get; set; }
    }

}

File: frontend/app/src/main/java/com/example/geoquiz_frontend/data/remote/dtos/koth/RoundStartedData.java
package com.example.geoquiz_frontend.data.remote.dtos.koth;

import com.example.geoquiz_frontend.domain.enums.RoundType;
import com.google.gson.annotations.SerializedName;

public class RoundStartedData {

    @SerializedName("roundNumber")
    private int roundNumber;

    @SerializedName("roundType")
    private int roundType;

    @SerializedName("question")
    private QuestionData question;

    @SerializedName("serverTime")
    private long serverTime;
    @SerializedName("roundEndAt")
    private long roundEndAt;

    public int getRoundNumber() { return roundNumber; }
    public int getRoundTypeValue() { return roundType; }
    public RoundType getRoundType() { return RoundType.fromValue(roundType); }
    public QuestionData getQuestion() { return question; }
    public long getServerTime() { return serverTime; }
    public long getRoundEndAt() { return roundEndAt; }
}